SupportYourLocals.Data/DataStorage.cs

```diff
@@ -21,17 +21,31 @@ public enum ProductType
 
     public class LocationData
     {
-        public int ID { get; set; }
+        public string ID { get; set; }
         public Location Location { get; set; }
         public string Name { get; set; }
         public int AddedByID { get; set; }
         public DateTime Time { get; set; }
         public Dictionary<ProductType, List<string> > Products { get; set; }
+
+        public LocationData(Location location, string name, int addedByID, DateTime time, Dictionary<ProductType, List<string>> products)
+        {
+            ID = GenerateId();
+            Location = location;
+            Name = name;
+            AddedByID = addedByID;
+            Time = time;
+            Products = products;
+        }
+        public string GenerateId()
+        {
+            return Guid.NewGuid().ToString("N");
+        }
```

Comment:
Maybe replace this with a lambda expression? would look cleaner.

```suggestion
        public string GenerateId() => Guid.NewGuid().ToString("N");
```